Add Elixir Security Live V2 Importer

michaelehab opened the pull request:
Solves #1933

* Update Elixir Security importer to filter and process advisories relevant to the purl passed in the constructor
* Update Elixir Security v2 importer to filter and process advisories relevant to the purl passed in the constructor
* Update Elixir Security importer tests to include testing package-first mode

TG1999 commented on vulnerabilities/importers/elixir_security.py:
```diff
@@ -30,7 +34,22 @@ class ElixirSecurityImporter(Importer):
+    def __init__(self, purl=None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.purl = purl
+        if self.purl:
```
This pattern is being used at multiple importers, we shall extract it out as a function

TG1999 replied:
We can always pass the message or the type or even the log as a parameter

michaelehab replied:
I like the idea of passing the supported types, that would make it generic with less duplicate code
The warning message is the same, so it will adapt if we pass supported types

keshav-space reviewed (changes requested):
@michaelehab there is no need to override `__init__` in pipeline.

michaelehab commented on vulnerabilities/pipelines/v2_importers/elixir_security_importer.py:
```diff
@@ -37,25 +41,65 @@ class ElixirSecurityImporterPipeline(VulnerableCodeBaseImporterPipelineV2):
     repo_url = "git+https://github.com/dependabot/elixir-security-advisories"
     unfurl_version_ranges = True
 
+    is_batch_run = True
+
+    def __init__(self, *args, purl=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.purl = purl
+        if self.purl:
+            ElixirSecurityImporterPipeline.is_batch_run = False
+            if self.purl.type != "hex":
+                self.log(
+                    f"Warning: PURL type {self.purl.type} is not 'hex', may not match any advisories"
+                )
+
```
And based on @TG1999's comment, we can unify the purl handling in package-first mode by overriding the VulnerableCodeBaseImporterPipeline constructor for example to do something like this.
```
def __init__(self, *args, **kwargs):
    super().__init__(*args, **kwargs)
    self.purl = kwargs.get("purl")
    self.supported_types = kwargs.get("supported_types)
```
Then doing the package-first checks and warnings which is unified for all importers.

michaelehab committed:
Add Elixir Security Live V2 Importer Pipeline #1933

* Add Elixir Security Live V2 Importer

* Add tests for the Elixir Security Live V2 Importer

* Tested functionally using the Live Evaluation API in #1969

Signed-off-by: Michael Ehab Mikhail <michael.ehab@hotmail.com>

ziadhany committed:
Migrate Elixir Security Importer live Pipeline
Update the Elixir Security Importer so we can have a separate function for parsing yaml file

Signed-off-by: ziad hany <ziadhany2016@gmail.com>
